packages/cli/internal/pkg/cli/format/format.go

```diff
@@ -4,24 +4,28 @@ import (
 	"os"
 	"text/tabwriter"
 )
+var Default Formatter
 
-var Formats = map[string]Formatter{
-	"text":    textWriter,
-	"tabular": tabularWriter,
-}
-var Format string
+type FormatterType string
+const (
+	textFormat FormatterType = "text"
+	tabularFormat FormatterType = "tabular"
+	DefaultFormat = textFormat
+)
 
-var textWriter Formatter = &Text{os.Stdout}
-var tabularWriter = &TextTabular{
-	*tabwriter.NewWriter(os.Stdout, 0, 8, 0, '\t', 0),
+var Formats = map[FormatterType]func() Formatter{
+	textFormat:    func() Formatter { return &Text{os.Stdout} },
+	tabularFormat: func() Formatter {
+		return &TextTabular{
+		*tabwriter.NewWriter(os.Stdout, 0, 8, 0, tabularDelimiter[0], 0),
+	}},
 }
-var Default Formatter
 
-func SetDefaultWriter() {
-	if writer, ok := Formats[Format]; ok {
-		Default = writer
+func SetFormatter(format FormatterType) {
```

Comment:
The code looks good. A few final things:
1. If no one calls SetFormatter then Default formatter is nil - not very safe
2. Usage of map here seems to make things a bit more hard to read.

Here is how we can improve it:

```go
var Default = NewText()

// in Text formatter file
func NewText() *Text {
  return &Text{os.Stdout} 
}

// in tabular formatter file
func NewTable () *Table {
  return &Table{tabwriter.NewWriter(os.Stdout, 0, 8, 0, tabularDelimiter[0], 0)}
}

func SetFormatter(format FormatterType) {
  switch format {
  case textFormat: Default = NewText()
  case tableFormatter: Default = NewTable()
  }
}
```